pkgs/by-name/ts/tsx/package.nix

```diff
@@ -0,0 +1,80 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  testers,
+  tsx,
+  pnpm_9,
+  nodejs_22,
+}:
+stdenv.mkDerivation rec {
+  pname = "tsx";
+  version = "4.19.3";
+
+  src = fetchFromGitHub {
+    owner = "privatenumber";
+    repo = "tsx";
+    tag = "v${version}";
+    hash = "sha256-wdv2oqJNc6U0Fyv4jT+0LUcYaDfodHk1vQZGMdyFF/E=";
+  };
+
+  pnpmDeps = pnpm_9.fetchDeps {
+    inherit pname version src;
+    hash = "sha256-57KDZ9cHb7uqnypC0auIltmYMmIhs4PWyf0HTRWEFiU=";
+  };
+
+  nativeBuildInputs = [
+    nodejs_22
+    pnpm_9.configHook
+  ];
+
+  buildInputs = [
+    nodejs_22
+  ];
+
+  # by default pnpm builds the docs workspace and this was just
+  #  the easiest way I found to stop that, as pnpmWorkspaces and
+  #  other flags did not work
+  patches = [ ./remove-docs.patch ];
+
+  buildPhase = ''
+    runHook preBuild
+
+    npm run build
+
+    # remove devDependencies that are only required to build
+    #  and package the typescript code
+    pnpm prune --prod
+
+    # Clean up broken symlinks left behind by `pnpm prune`
+    # https://github.com/pnpm/pnpm/issues/3645
+    find node_modules -xtype l -delete
+
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/{bin,lib/tsx}
+    cp -r {dist,node_modules} $out/lib/tsx
+    chmod +x $out/lib/tsx/dist/cli.mjs
+    ln -s $out/lib/tsx/dist/cli.mjs $out/bin/tsx
+
+    runHook postInstall
+  '';
+
+  passthru.tests = {
+    version = testers.testVersion {
+      package = tsx;
```

Comment:
Use versionCheckHook instead

Reply:
Done. Tried to run regular test suite but it had too many failures for me to understand while no broken functionality